flytekit/core/type_engine.py

```diff
@@ -1673,7 +1673,10 @@ async def async_to_python_value(  # type: ignore
             return self.from_binary_idl(lv.scalar.binary, expected_python_type)  # type: ignore
 
         try:
-            lits = lv.collection.literals
+            if lv and lv.scalar and lv.scalar.union.value.collection.literals:
+                lits = lv.scalar.union.value.collection.literals
```

Comment:
can you help me add more comments to let others know what this is for?
I think you might be right, but why this is related to `union` but not `literalmap` something like that?

Reply:
I am not sure, but I think it is because, in the first example above, the output of `FooTask` will be defined in `Interface`, so this will be related to `union`
<img width="694" alt="image" src="https://github.com/user-attachments/assets/6d8a3631-5af6-4b65-8d0b-651a414c9b65" />

After I commented the `has_foos, foos = foo_task()`, It will not be related to `union`.
<img width="682" alt="image" src="https://github.com/user-attachments/assets/9bc12ac9-7266-4b60-b982-363f1239829e" />